app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt

```diff
@@ -103,16 +103,18 @@ class TrackingProtectionFragment : PreferenceFragmentCompat() {
             getString(R.string.preference_enhanced_tracking_protection_strict_info_button)
         radioStrict.onPreferenceChangeListener = object : SharedPreferenceUpdater() {
             override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
-                if (newValue == true) {
-                    updateTrackingProtectionPolicy()
-                    context?.metrics?.track(
-                        Event.TrackingProtectionSettingChanged(
-                            Event.TrackingProtectionSettingChanged.Setting.STRICT
+                return super.onPreferenceChange(preference, newValue).also {
```

Comment:
@Baron-Severin @Amejia481  I moved  ` updateTrackingProtectionPolicy()` after the preference change, like Severin recommended for the checkboxes. Also, I've debugged this and with previous setting all the other preference keys remained true even if visually the radioButton was unchecked. That is why I added the `setCheckedWithoutClickListener` call. Any thoughts?